crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -834,11 +875,33 @@ pub fn get_child_id_from_topic(topic: &str) -> Result<Option<String>, Conversion
     }
 }
 
+pub fn get_child_id_from_event_topic(topic: &str) -> Result<Option<String>, ConversionError> {
+    match topic.strip_prefix("tedge/events/").map(String::from) {
+        Some(child_ev_topic) => {
+            let v: Vec<&str> = child_ev_topic.splitn(2, '/').collect();
+            if v.len() >= 2 {
+                let maybe_child_id = v[1].to_string();
+                if !maybe_child_id.is_empty() {
+                    return Ok(Some(maybe_child_id));
+                } else {
+                    return Err(ConversionError::InvalidChildId { id: maybe_child_id });
+                }
+            }
+            Ok(None)
+        }
+        None => Ok(None),
```

Comment:
It's not Ok! If `topic.strip_prefix("tedge/events/")` returns `None`, that means the input topic name does not have `tedge/events/`, but this function is called! It's error.